src/spikeinterface/preprocessing/decimate.py

```diff
@@ -0,0 +1,133 @@
+import numpy as np
+from spikeinterface.core.core_tools import (
+    define_function_from_class,
+)
+
+from .basepreprocessor import BasePreprocessor
+from .filter import fix_dtype
+from ..core import BaseRecordingSegment
+
+
+class DecimateRecording(BasePreprocessor):
+    """
+    Decimate the recording extractor traces using array slicing
+
+    Important: This uses simple array slicing for decimation rather than eg scipy.decimate.
+    This might introduce aliasing, or skip across signal of interest.
+    Consider  spikeinterface.preprocessing.ResampleRecording for safe resampling.
+
+    Parameters
+    ----------
+    recording : Recording
+        The recording extractor to be decimated.
+    decimation_frame_step : int
+        Step between successive frames sampled from the parent recording.
+    decimation_frame_start : int, default: 0
+        Index of first frame sampled from the parent recording.
+
+    Returns
+    -------
+    decimate_recording: DecimateRecording
+        The decimated recording extractor object. The full traces of the child recording correspond
+        to the parent traces as follows:
+            ```<decimated_traces> = <parent_traces>[<decimation_frame_start>::<decimation_frame_step>]```
+
+    """
+
+    name = "decimate"
+
+    def __init__(
+        self,
+        recording,
+        decimation_frame_step,
+        decimation_frame_start=0,
+    ):
+        # Original sampling frequency
+        self._orig_samp_freq = recording.get_sampling_frequency()
+        if not isinstance(decimation_frame_step, int) or decimation_frame_step <= 0:
+            raise ValueError(f"Expecting strictly positive integer for `decimation_frame_step` arg")
+        self._decimation_frame_step = decimation_frame_step
+        if not isinstance(decimation_frame_start, int) or decimation_frame_step < 0:
+            raise ValueError(f"Expecting positive integer for `decimation_frame_step` arg")
+        self._decimation_frame_start = decimation_frame_start
+        resample_rate = self._orig_samp_freq / self._decimation_frame_step
+        self._sampling_frequency = resample_rate
+        # fix_dtype not always returns the str, make sure it does
+        dtype = fix_dtype(recording, None).str
```

Comment:
You can remove this too, since it's inherited from the parent by default in the `BasePreprocessor`
```suggestion
```